thingooConnector/httpconnector.py

```diff
@@ -0,0 +1,176 @@
+
+import requests
+
+from thingooConnector import config
+from thingooConnector.config import REGISTER_ENDPOINT
+from thingooConnector.connector import Connector
+from thingooConnector.encoder import ComplexEncoder
+
+logger = logging.getLogger(__name__)
+
+
+class ClientCredentials:
+    """
+    A class used for store client credentials which are used to get access token
+    """
+
+    def __init__(self, client_id, client_secret):
+        self._client_id = client_id
+        self._client_secret = client_secret
+
+    def client_id(self):
+        return self._client_id
+
+    def client_secret(self):
+        return self._client_secret
+
+
+class HTTPException(Exception):
+    def __init__(self, error_text, status_code, text):
+        super().__init__(f'{error_text}: {status_code} {text}')
+
+
+class RetrieveTokenException(HTTPException):
+    def __init__(self, status_code, text):
+        super().__init__("Fail to retrieve access token", status_code, text)
+
+
+class RegisterDeviceException(HTTPException):
+    def __init__(self, status_code, text):
+        super().__init__("Fail to register device", status_code, text)
+
+
+class Token:
+    """
+    A class used for store token
+    """
+
+    def __init__(self, json):
+        """
+        Create token from json
+        :param json: A dict with token fields
+        """
+        self._access_token = json["access_token"]
+        self._expires_in = json["expires_in"]
+
+    def access_token(self):
+        return self._access_token
+
+
+class HTTPConnector(Connector):
+    """
+    Connector class to Thingoo instance
+    """
+
+    def __init__(self, device_info, host, client_credentials, entities):
+        super().__init__(host, device_info, entities)
+        self._client_credentials = client_credentials
+        self._token = None
+
+    def connect(self):
+        """
+        Connect to Thingoo instance: get token and register device
+        """
+        logger.info(f'Connecting to {self._host}...')
+        # Get OAuth token or raise RetrieveTokenException
+        self._token = self._get_token()
+        # Register device or raise RegisterDeviceException
+        self._register()
+        logger.info(f'Device connected!')
+
+    def _update_token(self):
+        """
+        Update token stored in self._token
+        """
+        try:
+            self._token = self._get_token()
+            logger.info("New OAuth token retrieved")
+        except RetrieveTokenException:
+            logger.warning("Fail to retrieve OAuth token")
+
+    def _get_token(self):
+        """
+        Get OAuth access token
+        :return: :class:`Token`
+        :raises: :class:`RetrieveTokenException`
+        """
+        request_data = {
+            "grant_type": "client_credentials",
+            "client_id": self._client_credentials.client_id(),
+            "client_secret": self._client_credentials.client_secret(),
+        }
+        url = f'https://{self._host}{REGISTER_ENDPOINT}'
```

Comment:
Rename REGISTER_ENDPOINT to, for example, TOKEN_ENDPOINT, as right now It's kind of misleading because of device registration.

Reply:
It was TOKEN_ENDPOINT but it is not possible due to the default Codacy configuration. I think that this check should be disabled.

Reply:
Change it to TOKEN_ENDPOINT and I will be able to ignore that one rule for this file then

Reply:
Resolved in 9134895f1006dbc6f966e64fff0160e3cade33cb